### ISSUE TITLE
Any possible ETA to support Helm 3?

### ISSUE THREAD

**Issue by bsethi** [flaper87/pyhelm#84]

**Comment by yanivoliver** [comment]
Since Helm 3 is now client only, I see two options for a possible pyHelm implementation:
- Reimplementing the Helm logic in Python
- Compiling Helm 3 as a shared library and performing Go calls from the pyHelm Python code

Both of these options sound complicated and generally very error prone. Given that, I don't see any implementation coming in the upcoming future.
Do you have any additional thoughts?

**Comment by kfox1111** [comment]
A few more options:
 * create a go grpc service wrapping the helm go library
 * exec helm

**Comment by jseadragon** [comment]
> Reimplementing the Helm logic in Python

This feels like a lot of maintenance chore to me. 

To me, it feels cleaner to go Python + Go bindings rather than `exec helm`, but I'm curious if there are other thoughts. 

For the `Go` bindings, 
  * Here's the client: https://github.com/helm/helm/blob/master/pkg/kube/client.go
  * Here's the `helm install` code. We could probably as a POC, just simply call this function (or re-implement it, it's only 200-300 lines of code anyway). https://github.com/helm/helm/blob/83847cc89980185ccbbd98747d7dc725e1e4910d/cmd/helm/install.go#L155

**Comment by pravicloud** [comment]
Hi Team,
Could you please provide an update for the helm3 -- pyhelm

**Comment by s-rathi** [comment]
Since there is no existing support for helm 3. Maybe you can use nodejs-helm.

GitHub: https://github.com/nikhilm98/nodejs-helm

npm-package: https://www.npmjs.com/package/nodejs-helm

You can run `npm install nodejs-helm` to install this packages.

**Comment by jseadragon** [comment]
My use case (and I suspect many others), would be for a native python solution. We wanted to use Helm with Ansible, as there are existing playbooks using the helm2 implementation. 

If you don’t actually need pythons maybe the nodejs would work. 

I’m curious about giving it a shot based on the approach mentioned earlier but I don’t know how much time I will have.

**Comment by zbrookle** [comment]
If anyone is still looking for a pythonic solution to installing / generating helm charts, I just released a package called [avionix]( https://github.com/zbrookle/avionix) which provides an object oriented solution to interacting with Kubernetes objects and Helm charts. My package does support helm3 which may make it useful to many of you.

**Comment by purnima995** [comment]
> If anyone is still looking for a pythonic solution to installing / generating helm charts, I just released a package called [avionix](https://github.com/zbrookle/avionix) which provides an object oriented solution to interacting with Kubernetes objects and Helm charts. My package does support helm3 which may make it useful to many of you.

Hi, this is great. I was thinking of giving it a try, can you please tell me how reliable is it? Can I use it on production level.

**Comment by zbrookle** [comment]
@purnima995 Yes! It's very reliable, I'm actually using it right now in production

**Comment by purnima995** [comment]
@zbrookle awesome. Thanks.

**Comment by eli-halych** [comment]
@flaper87 Any updates on this?

**Comment by flaper87** [comment]
> @flaper87 Any updates on this?

I'm not actively maintaining this package right now due to a lack of time. I'd be happy to hand it off to someone interested in working on it.

I'm sorry :cry:

**Comment by heisaman** [comment]
@flaper87 I'd like to work on it in my spare time. Just not sure does it still worth it as in Aug. 2022?

**Comment by flaper87** [comment]
> @flaper87 I'd like to work on it in my spare time. Just not sure does it still worth it as in Aug. 2022?

@heisaman thanks for your interest. I haven't used this package in while and I'm unsure on how active it is. Helm stil exists, and helm3 is widely adopted so, unless there's a better alternative, I would say revisiting this package may still be worth it.

**Comment by robcxyz** [comment]
FYI -> https://github.com/stackhpc/pyhelm3

Probably worth a PR to README to point to that repo and archive this one @flaper87 ?
